linux/guest_customization/check_network_config.yml

```diff
@@ -122,82 +132,116 @@
           ansible.builtin.set_fact:
             src_network_file: "/etc/network/interfaces.d/50-cloud-init"
           when: guest_os_ansible_distribution == "Debian"
-      when: "gosc_workflow == 'cloud-init'"
-  when: >
-    (guest_os_ansible_distribution == "Debian") or
-    (guest_os_ansible_distribution == "Ubuntu" and
-     guest_os_ansible_distribution_ver is version('19.10', '<') and
-     guest_os_with_gui is defined and guest_os_with_gui | bool)
 
-- name: "Set GOSC network configuration files on RedHat"
-  ansible.builtin.set_fact:
-    src_network_file: "/etc/sysconfig/network-scripts/ifcfg-{{ guest_iface_name }}"
+- name: "Collect network configuration file for {{ vm_guest_os_distribution }}"
   when: guest_os_family == "RedHat"
+  block:
+    - name: "Set GOSC network configuration file on {{ vm_guest_os_distribution }}"
+      ansible.builtin.set_fact:
+        src_network_file: /etc/sysconfig/network-scripts/ifcfg-{{ guest_iface_name }}
+      when: guest_os_network_manager != "NetworkManager"
+
+    - name: "Set GOSC network configuration file on {{ vm_guest_os_distribution }}"
+      when: guest_os_network_manager == "NetworkManager"
+      block:
+        - name: "List network configuration files on {{ vm_guest_os_distribution }}"
+          include_tasks: ../../common/vm_shell_in_guest.yml
+          vars:
+            vm_shell_cmd: "/usr/bin/nmcli"
+            vm_shell_args: "-t -f DEVICE,FILENAME connection show --active"
+            vm_shell_out: "/tmp/nmcli_connection_show.txt"
+
+        - name: "Fetch the result of listing network configuration files on {{ vm_guest_os_distribution }}"
+          include_tasks: ../../common/vm_guest_file_operation.yml
+          vars:
+            operation: "fetch_file"
+            src_path: "/tmp/nmcli_connection_show.txt"
+            dest_path: "{{ current_test_log_folder }}/nmcli_connection_show.txt"
+
+        - name: "Get network connection on {{ vm_guest_os_distribution }}"
+          ansible.builtin.set_fact:
+            guest_network_connection: "{{ lookup('file', current_test_log_folder ~ '/nmcli_connection_show.txt').split('\n') |
+                                          select('match', guest_iface_name) }}"
+
+        - name: "Get network configuration file  on {{ vm_guest_os_distribution }}"
+          ansible.builtin.set_fact:
+            src_network_file: "{{ guest_network_connection[0].split(':')[-1] }}"
+          when: guest_network_connection | length == 1
 
-- name: "Set GOSC network configuration files on SLE"
+- name: "Set GOSC network configuration file on {{ vm_guest_os_distribution }}"
   ansible.builtin.set_fact:
     src_network_file: "/etc/sysconfig/network/ifcfg-{{ guest_iface_name }}"
   when: guest_os_family == "Suse"
 
-- block:
-    - name: "Print the network configuration file on {{ guest_os_ansible_distribution }}"
+- name: "Collect network configuration file on {{ vm_guest_os_distribution }}"
+  when:
+    - src_network_file is defined
```

Comment:
What if `src_network_file` is undefined?

Reply:
For example, Photon OS doesn't define this variable, its network files are  under /etc/systemd/network/, which will be collected in below tasks.

Reply:
src_network_file is collected only for triage. If there is GOS has undefined and there is an GOSC issue and then we can came here to find what is missing. If there is no GOSC issue, nothing will be affected because IP, netmask and gateway have been retrieved in above tasks.